ReactSkia/components/RSkComponentImage.cpp

```diff
@@ -388,20 +394,25 @@ void RSkComponentImage::requestNetworkImageData(ImageSource source) {
     auto responseCacheControlData = responseData->headerBuffer.find("Cache-Control");
     if(responseCacheControlData != responseData->headerBuffer.items().end()) {
       std::string responseCacheControlString = responseCacheControlData->second.asString();
-      canCacheData_ = shouldCacheData(responseCacheControlString);
+      canCacheData_ = remoteCurlRequest->shouldCacheData();
       if(canCacheData_) responseMaxAgeTime = getCacheMaxAgeDuration(responseCacheControlString);
     }
 
     // TODO : Parse request headers and retrieve caching details
-
-    cacheExpiryTime_ = std::min(responseMaxAgeTime,static_cast<double>(DEFAULT_MAX_CACHE_EXPIRY_TIME));
+    cacheExpiryTime_ = std::min(CONVERT_SEC_TO_MS(responseMaxAgeTime),static_cast<double>(DEFAULT_MAX_CACHE_EXPIRY_TIME));
     RNS_LOG_DEBUG("url [" << responseData->responseurl << "] canCacheData[" << canCacheData_ << "] cacheExpiryTime[" << cacheExpiryTime_ << "]");
     return 0;
   };
 
 
   // completioncallback lambda fuction
-  auto completionCallback =  [this, remoteCurlRequest](void* curlresponseData,void *userdata)->bool {
+  auto completionCallback =  [this, weakThis = this->weak_from_this(), remoteCurlRequest](void* curlresponseData,void *userdata)->bool {
+    auto isAlive = weakThis.lock();
+    if(!isAlive) {
+      RNS_LOG_WARN("This object is already destroyed. ignoring the completion callback");
```

Comment:
need to reset the completioncallback and headercallback handles before return here ?